CMakeLists.txt

```diff
@@ -259,6 +259,9 @@ if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
   message(FATAL_ERROR "In-source build attempted; please clean the CMake cache and then switch to an out-of-source build, e.g.,\nrm CMakeCache.txt && rm -Rf CMakeFiles/\nmkdir build/ && cd build/ && cmake ..")
 endif()
 
+if (NOT CMAKE_BUILD_TYPE)
```

Comment:
It is also important that we handle mis-set build types (e.g., the old `PureDebug` choice should cause an error). I think that we need to enumerate the allowed values and complain otherwise.